x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -127,11 +54,21 @@ export class CloudPlugin implements Plugin<CloudSetup> {
       is_elastic_staff_owned: isElasticStaffOwned,
     } = this.config;
 
+    let decodedId: DecodedCloudId | undefined;
+    if (id) {
+      decodedId = decodeCloudId(id);
+    }
+
     return {
       cloudId: id,
+      deploymentId: parseDeploymentIdFromDeploymentUrl(this.config.deployment_url),
```

Comment:
Added the `deploymentId` (which was only exposed on the server-side contract) while I was at it, given some plugins recoded their helper to access it.